idea/gear/frontend/src/features/code-verifier/api/hooks.ts

```diff
@@ -0,0 +1,54 @@
+import { HexString } from '@gear-js/api';
+import { useAlert } from '@gear-js/react-hooks';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { STATUS_CODES } from 'http';
+import { useEffect } from 'react';
+
+import { getVerificationStatus, getVerifiedCode, verifyCode } from './requests';
+
+function useVerifyCode() {
+  return useMutation({
+    mutationKey: ['verify-code'],
+    mutationFn: verifyCode,
+  });
+}
+
+function useIsCodeVerified(codeId: HexString | null | undefined) {
+  const alert = useAlert();
```

Comment:
I think it can be worth to have some kind of cache for verified code ids just to avoid making unnecessary requests

Reply:
We're using `@tanstack/react-query` lib to handle async states, caching requests is part of it's functionality.